tests/integration/Owncloud/OcisPhpSdk/DriveTest.php

```diff
@@ -575,4 +575,307 @@ public function testGetMyDriveForDriveShareReceiver(): void
+            }
+        }
+
+        if (empty($viewerRole)) {
+            throw new \Error(
+                "Space viewer role not found "
+            );
+        }
+        // ocis stable doesn't support root endpoint
+        if (getenv('OCIS_VERSION') === "stable") {
+            $this->expectException(EndPointNotImplementedException::class);
+            $this->expectExceptionMessage("This method is not implemented in this ocis version");
+            $this->drive->invite($marie, $viewerRole);
+        }
+
+        try {
+            $this->drive->invite($marie, $viewerRole);
+            $marieReceivedProjectDrive = $marieOcis->getDriveById($this->drive->getId());
+            $this->expectException(ForbiddenException::class);
+            $this->expectExceptionMessage("accessDenied - add grant: error: permission denied:");
+            $marieReceivedProjectDrive->invite($katherine, $viewerRole);
+
+        } catch(EndPointNotImplementedException) {
+            // test should fail if ocis version is less than 6.0.0
+            $this->fail("EndPointNotImplementedException was thrown unexpectedly");
+        }
+    }
+
+    /**
+     * @throws ForbiddenException
+     * @throws InvalidResponseException
+     * @throws BadRequestException
+     * @throws EndPointNotImplementedException
+     * @throws UnauthorizedException
+     * @throws HttpException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     * @throws \Exception
+     */
+    public function testReceiverUpdatesDriveShareRole(): void
+    {
+        $marieOcis = $this->initUser('marie', 'radioactivity');
+
+        $marie = $this->ocis->getUsers('marie')[0];
+
+        $managerRole = null;
+
+        if (getenv('OCIS_VERSION') === "stable") {
+            // ocis version < 6.0.0 doesn't support root endpoint so mocking getRoles values
+            $role = [
+                "id" => "312c0871-5ef7-4b3a-85b6-0e4074c64049",
+                "description" => "Allows managing a space",
+                "displayName" => "Manager",
+                "@libre.graph.weight" => 3
+            ];
+
+            $shareRoles = [new SharingRole(new UnifiedRoleDefinition($role))];
+        } else {
+            $shareRoles = $this->drive->getRoles();
+        }
+        foreach ($shareRoles as $role) {
+            if ($role->getId() === self::getPermissionsRoleIdByName('Manager')) {
+                $managerRole = $role;
+                break;
+            }
+        }
+
+        if (empty($managerRole)) {
+            throw new \Error(
+                "manager role not found "
+            );
+        }
+        // ocis stable doesn't support root endpoint
+        if (getenv('OCIS_VERSION') === "stable") {
+            $this->expectException(EndPointNotImplementedException::class);
+            $this->expectExceptionMessage("This method is not implemented in this ocis version");
+            $this->drive->invite($marie, $managerRole);
+        }
```

Comment:
may be we can skip this part

Reply:
Yes, if it's tested somewhere else there is no point to copy it into every test

Reply:
You will still need to do something to handle what happens when the unit test suite is run on old ocis. At least mark test skipped.